satpy/readers/iasi_l2.py

```diff
@@ -105,62 +113,69 @@ def get_dataset(self, key, info):
         with h5py.File(self.filename, 'r') as fid:
             LOGGER.debug('Reading %s.', key.name)
             if key.name in DSET_NAMES:
-                m_data = read_dataset(fid, key, info)
+                m_data = read_dataset(fid, key)
             else:
-                m_data = read_geo(fid, key, info)
-        m_data.info.update(info)
+                m_data = read_geo(fid, key)
+        m_data.attrs.update(info)
 
         return m_data
 
 
-def read_dataset(fid, key, info):
+def read_dataset(fid, key):
     """Read dataset"""
     dsid = DSET_NAMES[key.name]
-    data = fid["/PWLR/" + dsid].value
+    dset = fid["/PWLR/" + dsid]
+    if dset.ndim == 3:
+        dims = ['y', 'x', 'level']
+    else:
+        dims = ['y', 'x']
+    data = xr.DataArray(da.from_array(dset.value, chunks=CHUNK_SIZE),
+                        name=key.name, dims=dims).astype(np.float32)
+    data = xr.where(data > 1e30, np.nan, data)
+
     try:
-        unit = fid["/PWLR/" + dsid].attrs['units']
-        long_name = fid["/PWLR/" + dsid].attrs['long_name']
+        dset_attrs = dict(dset.attrs)
     except KeyError:
-        unit = ''
-        long_name = ''
+        dset_attrs = {}
 
-    data = np.ma.masked_where(data > 1e30, data)
+    data.attrs.update(dset_attrs)
 
-    return Dataset(data, copy=False, long_name=long_name,
-                   **info)
+    return data
 
 
-def read_geo(fid, key, info):
+def read_geo(fid, key):
     """Read geolocation and related datasets."""
     dsid = GEO_NAMES[key.name]
+    add_epoch = False
     if "time" in key.name:
         days = fid["/L1C/" + dsid["day"]].value
         msecs = fid["/L1C/" + dsid["msec"]].value
-        unit = ""
         data = _form_datetimes(days, msecs)
+        add_epoch = True
     else:
         data = fid["/L1C/" + dsid].value
-        unit = fid["/L1C/" + dsid].attrs['units']
 
-    data = Dataset(data, copy=False, **info)
+    data = xr.DataArray(da.from_array(data, chunks=CHUNK_SIZE),
+                        name=key.name, dims=['y', 'x']).astype(np.float32)
+    if add_epoch:
+        data.attrs['sensing_time_epoch'] = EPOCH
 
     return data
 
 
 def _form_datetimes(days, msecs):
-    """Form datetimes from days and milliseconds relative to EPOCH for
-    each of IASI scans."""
+    """Calculate seconds since EPOCH from days and milliseconds for each of IASI scan."""
 
     all_datetimes = []
     for i in range(days.size):
         day = int(days[i])
         msec = msecs[i]
         scanline_datetimes = []
-        for j in range(VALUES_PER_SCAN_LINE / 4):
+        for j in range(int(VALUES_PER_SCAN_LINE / 4)):
```

Comment:
Is this ever going to be not a whole number? I suppose the `int` ends up flooring it anyway.

Reply:
I added the call to `int()` as `range()` doesn't allow the argument to be a float, which it is for every (single-`/`) division in Python 3. For Python 2 the call isn't needed.